### PR TITLE
add morph support

### PR DESCRIPTION
**Pull request by rendyananta**
This pull requests consists of

1. Morphable support
   1. Every entry must have an author
   1. Every journal can have a recordable

2. Refactoring
3. Add missing code documentations

### CODE REVIEW

File: database/migrations/2020_10_30_000002_create_journals_table.php [hunk 1]
```diff
@@ -18,7 +18,12 @@ public function up()
             $table->unsignedBigInteger('amount')->default(0);
             $table->string('memo')->nullable();
             $table->string('ref')->nullable();
+            $table->string('recordable_type')->nullable();
```
Comment (addeeandra) [inline database/migrations/2020_10_30_000002_create_journals_table.php:21]: Any reason not to use `nullableMorph('recordable')`?

File: src/Contracts/EntryAuthor.php [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    function entries(): MorphMany;
+
+    /**
+     * Create single entry on specific journal
+     *
+     * @param Journal $journal
+     * @param $entry
+     * @return \Illuminate\Database\Eloquent\Model
+     */
+    function createEntry(Journal $journal, $entry);
```
Comment (addeeandra) [inline src/Contracts/EntryAuthor.php:25]: The `$entry` argument should strictly defined as `Entry` interface

File: src/Concerns/HasEntries.php [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    }
+
+    /**
+     * Create single entry on specific journal
+     *
+     * @param Journal $journal
+     * @param $entry
+     * @return \Illuminate\Database\Eloquent\Model
+     */
+    function createEntry(Journal $journal, $entry)
```
Comment (addeeandra) [inline src/Concerns/HasEntries.php:31]: This `$entry` argument should strictly defined as `Entry` interface

**Review by addeeandra — CHANGES_REQUESTED** [review]
LGTM! -- Please see some of my reviews

**Review by addeeandra — COMMENTED** [review]
LGTM!

**Comment by rendyananta** [comment]
Well, I have been implementing your request of changes.

**Review by addeeandra — APPROVED** [review]

**Pull request merged**